scripts/pylib/twister/twisterlib/runner.py

```diff
@@ -27,29 +27,61 @@
 
 class ExecutionCounter(object):
     def __init__(self, total=0):
+        '''
+        Most of the stats are at test instance level
+        Except that "_cases" and "_skipped_cases" are for cases of ALL test instances
+
+        total complete = done + skipped_filter
+        total = yaml test scenarios * applicable platforms
+        complete perctenage = (done + skipped_filter) / total
+        pass rate = passed / (total - skipped_configs)
+        '''
+        # instances that go through the pipeline
+        # updated by report_out()
         self._done = Value('i', 0)
+
+        # instances that actually executed and passed
+        # updated by report_out()
         self._passed = Value('i', 0)
+
+        # static filter + runtime filter + build skipped
+        # updated by update_counting_before_pipeline() and report_out()
         self._skipped_configs = Value('i', 0)
+
+        # cmake filter + build skipped
+        # updated by report_out()
         self._skipped_runtime = Value('i', 0)
+
+        # staic filtered at yaml parsing time
+        # updated by update_counting_before_pipeline()
         self._skipped_filter = Value('i', 0)
+
+        # updated by update_counting_before_pipeline() and report_out()
         self._skipped_cases = Value('i', 0)
+
+        # updated by report_out() in pipeline
         self._error = Value('i', 0)
         self._failed = Value('i', 0)
+
+        # initialized to number of test instances
         self._total = Value('i', total)
+
+        # updated in update_counting_after_pipeline()
         self._cases = Value('i', 0)
         self.lock = Lock()
 
     def summary(self):
         logger.debug("--------------------------------")
-        logger.debug(f"Total Test suites: {self.total}")
-        logger.debug(f"Total Test cases: {self.cases}")
+        logger.debug(f"Total test instances: {self.total}")
+        logger.debug(f"Total test cases: {self.cases}")
+        logger.debug(f"Executed test cases: {self.cases - self.skipped_cases}")
         logger.debug(f"Skipped test cases: {self.skipped_cases}")
-        logger.debug(f"Completed Testsuites: {self.done}")
-        logger.debug(f"Passing Testsuites: {self.passed}")
-        logger.debug(f"Failing Testsuites: {self.failed}")
-        logger.debug(f"Skipped Testsuites: {self.skipped_configs}")
-        logger.debug(f"Skipped Testsuites (runtime): {self.skipped_runtime}")
-        logger.debug(f"Skipped Testsuites (filter): {self.skipped_filter}")
+        logger.debug(f"Completed test instances: {self.done}")
```

Comment:
instance is internal to twister, we should keep this as testsuite, because this is exactly what will show up in the test report, a testsuite.

Reply:
Didn't realize that we want to encapsulate the concept of "test instance".
Will change it.

Reply:
But the stats like `skipped_configs` are all calculated at the test instance level. Such as below:
https://github.com/zephyrproject-rtos/zephyr/blob/d43e4e9ca4e80055e9a0f0b095c856103ee0d11e/scripts/pylib/twister/twisterlib/runner.py#L602-L604

I can change back to the `testsuite` in the log message. But it's not really what the numbers mean.

Reply:
still, instance is historical and internal to twister. Please just keep it as is and stick to the original purpose of the PR.

Reply:
@nashif 

Got it. Just re-pushed.